lbz/events/api.py

```diff
@@ -97,6 +94,7 @@ def _create_eb_entry(self, new_event: Event) -> PutEventsRequestEntryTypeDef:
 
 def event_emitter(function: Callable) -> Callable:
     """Decorator that makes function an emitter - automatically sends pending events on success"""
+    EventAPI().clear()
```

Comment:
What about the below clearing? 🤔 Of course, we should get rid of all the pending events but the current logic says that failed/sent events should be kept during the whole "session".

Reply:
It looks like you updated the `EventAwareResource` class but not the `event_emitter` decorator.